pull.go

```diff
@@ -53,27 +53,17 @@ func localImageNameForReference(store storage.Store, srcRef types.ImageReference
 	return name, nil
 }
 
-func pullImage(store storage.Store, options BuilderOptions, sc *types.SystemContext) (types.ImageReference, error) {
-	name := options.FromImage
-
-	spec := name
-	if options.Registry != "" {
-		spec = options.Registry + spec
-	}
-	spec2 := spec
-	if options.Transport != "" {
-		spec2 = options.Transport + spec
-	}
-
-	srcRef, err := alltransports.ParseImageName(name)
+func pullImage(store storage.Store, imageName string, options BuilderOptions, sc *types.SystemContext) (types.ImageReference, error) {
+	spec := imageName
```

Comment:
could the imageName be just the ID at this point?  If so I think we still end up with a srcRef that has that id in it rather than converting it the right one.   i.e.:

docker.io/library/e87186c3d14:latest

rather than 
docker.io/anarcat/debian-airsonic:latest

I had added a 'importBuilderFromImage() call here and that converted the srcRef appropriately for me, but I was trying to figure out how to get the code to recognize that the image was stored locally instead of the original pull location at github.

Reply:
It shouldn't be - if it matches an image that we have locally, we'll skip attempts to search registries for it.  Otherwise, we'll fall through and try to treat the ID as a name, but I don't think the registry protocol lets us pull an image without a name.

Reply:
I just took another look.  What I was concerned about is if  you passed in the id (e87186c3d14) into the call for ResolveName() it would just find the long version of that id and potentially send that back to pullAndFindImage() which would set srcRef to the image id.  But that only happens in newBuilder(), I thought elsewhere, so that should be OK and we're probably not passing in an ID at that point anyway.